core/Connection.js

```diff
@@ -0,0 +1,25 @@
+var FIFO = require('./FIFO');
+
+var Connection = function () {
+  this.closed = false;
+  this.contents = new FIFO();
+  this._runtime = null;
```

Comment:
Connections are really the workhorses of IP transfer (rather than ports). They're the ones that modify process status and interact with the runtime, so I've moved this here.

I also made this base Connection object.